src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessThreadTests.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Security;
+using System.IO;
+using Xunit;
+
+namespace System.Diagnostics.ProcessTests
+{
+    public class ProcessThreadTests : ProcessTestBase
+    {
+        [Fact]
+        public void TestCommonPriorityAndTimeProperties()
+        {
+            ProcessThreadCollection threadCollection = _process.Threads;
+            Assert.True(threadCollection.Count > 0);
+
+            ProcessThread thread = threadCollection[0];
+
+            if (ThreadState.Terminated != thread.ThreadState)
+            {
+                Assert.Equal(_process.BasePriority, thread.BasePriority);
+                Assert.True(thread.CurrentPriority >= 0, "Thread CurrentPriority");
+                Assert.True(thread.PrivilegedProcessorTime.TotalSeconds >= 0, "Thread PrivilegedProcessorTime");
+                Assert.True(thread.UserProcessorTime.TotalSeconds >= 0, "Thread UserProcessorTime");
+                Assert.True(thread.TotalProcessorTime.TotalSeconds >= 0, "Thread TotalProcessorTime");
+            }
+        }
+
+        [Fact]
+        public void TestStartTimeProperty()
+        {
+            ProcessThreadCollection threadCollection = _process.Threads;
+            Assert.True(threadCollection.Count > 0);
+
+            ProcessThread thread = threadCollection[0];
+
+            if (global::Interop.IsOSX)
+            {
+                Assert.Throws<PlatformNotSupportedException>(() => thread.StartTime);
+                return;
+            }
+
+            if (ThreadState.Initialized != thread.ThreadState)
+            {
+                Assert.True(thread.StartTime.ToUniversalTime() <= DateTime.UtcNow, "Thread StartTime");
```

Comment:
I tried to do that for process starttime test, but somehow the Starttime of the process was lower than the datetime.now instance taken before process start on Mac. On Mac we get the number of nanoseconds elapsed from start of process to datetime.now (i got this data from code comment, couldn't find this documented anywhere @sokket can talk about the source), maybe that is why we almost hit the same value. So the safest way to test was to compare with a later time. I can test with a prior time by making thread.sleep(1) to ensure there is some lag between datetime.now and process.start() instructions..